scripts/cmake-configurator/parser/cmake_parser.py

```diff
@@ -0,0 +1,48 @@
+from cmake_row import cmake_row
+from cmake_variable import cmake_variable
+from typing import List
+
+class cmake_parser:
+   def read_file(self, abs_file_path: str) -> None:
+      "reads file lines and inserts them into file_lines"
+      self.file_lines: List[str] = []
+      with open(abs_file_path) as f:
+         self.file_lines = f.readlines()
+
+   def lines_to_vars(self,file_lines: List[str]) -> List[cmake_variable]:
+      """
+      this will communicate with the gui business logic, and pass to it
+      variables ready to be displayed
+      """
+      self.file_lines: List[str] = file_lines
+      cmake_rows :List[cmake_row] =  []
+      cmake_vars :List[cmake_variable] = []
+
+      for i, line in enumerate(self.file_lines):
+         cmake_rows.append(cmake_row(line, i))
+
+      for i, row in enumerate(cmake_rows):
+         if row.type == cmake_row.VARIABLE:
+            comment: str = ""
+            j = i-1
+            while j >= 0 and cmake_rows[j].type == cmake_row.SLASH_COMMENT:
+               comment = cmake_rows[j].val + comment
+               j -=1
+            cmake_vars.append(cmake_variable(row, comment))
```

Comment:
That's cool, but we're saving the row-comments as they are, without any post-processing.
We should (now, or in the next commit) strip the "//" and append all comment lines into a single line. Keeping the comment as it is now, it doesn't make much sense.

Reply:
This is already done by `cmake_row`'s constructor

Reply:
Ah, OK, that's great! I just didn't notice it.